### PR TITLE
Refactor accounts-identity UI

### PR DESCRIPTION
**Pull request by zarvox**
This refactor makes explicit all of the states that the UI can be in.  We also
handle any Meteor method call failures, displaying the error message in the UI.
This will allow us to add UI to explain why login with identity fails when a user limit is reached, and provides a reasonable place for us to prompt for two-factor auth in the more distant future.

In the non-login-identity view, we:

* split the identity list out by account with a faint horizontal rule
* retain only one "unlink" button per account, since unlinking is an action
  taken on the account, rather than on the identity pictured
* avoid using window.confirm() in favor of an in-app modal dialog
* visibly handle success/failure of any of the Meteor method calls with
  the focusing success/error boxes

Additional stylistic changes include avoiding using the `event` global,
using `evt.currentTarget` rather than `evt.target`, and preferring ES6 fat-arrow
functions in most places.

non-login-identity

no-window-confirm

### CODE REVIEW

**Comment by zarvox** [comment]
Looks like JSCS can't handle object spreads, which is actually somewhat reasonable because I think they're officially an ES7 thing.

**Comment by kentonv** [comment]
This seems like a good one for @dwrensha to review.

**Comment by dwrensha** [comment]
> retain only one "unlink" button per account, since unlinking is an action taken on the account, rather than on the identity pictured

nice!

**Comment by kentonv** [comment]
(@dwrensha let me know when you think this is ready for merge.)

File: shell/packages/accounts-identity/accounts-identity-client.js [hunk 1]
```diff
@@ -30,14 +30,61 @@ Accounts.isLinkingNewIdentity = function () {
...
+  //
+  //   * loading                 Waiting on subscriptions to finish loading
+  //   * creatingAccount         We have sent a createAccountForIdentity call and are awaiting the reply
+  //   * accountCreationFailed   The createAccountForIdentity call failed.  Value is the Error.
+  //   * loggingInWithIdentity   We have sent a loginWithIdentity call and are awaiting the reply
+  //   * loginWithIdentityFailed The loginWithIdentity call failed.  Value is the Error.
+  //   * loggingInWithToken      We have started logging in with a token and are awaiting the reply
+  //   * loginWithTokenFailed    The loginWithToken call failed.  Value is the Error.
+  //   * linkingIdentity         We have sent a linkIdentityToAccount call and are awaiting the reply
+  //   * noLoginIdentities       Subscriptions loaded, but we can neither login, create, nor link an account.  We require input.
```
Comment (dwrensha) [inline shell/packages/accounts-identity/accounts-identity-client.js]: Would `identityCannotLogin` be a more appropriate name for this state?

File: shell/packages/accounts-identity/accounts-identity-client.js [hunk 2]
```diff
@@ -30,14 +30,61 @@ Accounts.isLinkingNewIdentity = function () {
...
+  //
+  // The noLoginIdentities state also tracks substates directly:
```
Comment (dwrensha) [inline shell/packages/accounts-identity/accounts-identity-client.js]: s/noLoginIdentities/unlinkIdentityState/`?

File: shell/packages/accounts-identity/accounts-identity-client.js (hunk 1 above)
Comment (zarvox) [inline shell/packages/accounts-identity/accounts-identity-client.js]: I might find that a bit ambiguous with `loginWithIdentityFailed`.

Maybe `identityIsNotLoginIdentity`?

File: shell/packages/accounts-identity/accounts-identity-client.js (hunk 2 above)
Comment (zarvox) [inline shell/packages/accounts-identity/accounts-identity-client.js]: Clarified in c3a8014

**Comment by zarvox** [comment]
Updated comments and such.

**Comment by dwrensha** [comment]
Looks good to me.

**Pull request merged**